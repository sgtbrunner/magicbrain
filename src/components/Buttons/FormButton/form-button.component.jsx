import React from 'react';
import PropTypes from 'prop-types';

const FormButton = ({ children, isValidForm, disabled }) => {
  const getIsValidFormClass = (isValidForm) => (isValidForm ? 'b bg-lightest-blue grow' : '');

  const isValidClass = getIsValidFormClass(isValidForm);

  return (
    <button
      className={`ph3 pv2 mt3 ba b--black f6 dib ${isValidClass}`}
      type="submit"
      disabled={disabled}
    >
      {children}
    </button>
  );
};

FormButton.propTypes = {
  isValidForm: PropTypes.bool,
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
  disabled: PropTypes.bool,
};

FormButton.defaultProps = {
  isValidForm: false,
  disabled: false,
};

export default FormButton;
