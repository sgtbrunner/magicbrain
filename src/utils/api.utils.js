const { REACT_APP_API_ENDPOINT } = process.env;

const fetchServer = ({
  path,
  method = 'POST',
  headers = { 'Content-Type': 'application/json' },
  body,
}) =>
  fetch(`${REACT_APP_API_ENDPOINT}${path}`, {
    method,
    headers,
    body: JSON.stringify({ ...body }),
  });

export default {
  signInUser: ({ email, password }) =>
    fetchServer({ path: '/signin', body: { email, password } }).then((response) => response.json()),

  registerUser: ({ name, email, password }) =>
    fetchServer({ path: '/register', body: { name, email, password } }).then((response) =>
      response.json()
    ),

  getImageBoundary: ({ input }) =>
    fetchServer({ path: '/imageurl', body: { input } }).then((response) => response.json()),

  updateImageCount: ({ id }) =>
    fetchServer({ path: '/image', method: 'PUT', body: { id } }).then((response) =>
      response.json()
    ),
};
