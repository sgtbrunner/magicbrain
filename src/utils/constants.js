export const PARTICLE_OPTIONS = {
  particles: {
    number: {
      value: 50,
      density: {
        enable: true,
        value_area: 500,
      },
    },
  },
};

export const { REACT_APP_API_ENDPOINT } = process.env;
