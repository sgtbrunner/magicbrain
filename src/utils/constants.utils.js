const PARTICLE_OPTIONS = {
  particles: {
    number: {
      value: 50,
      density: {
        enable: true,
        value_area: 500,
      },
    },
  },
};

export default PARTICLE_OPTIONS;
