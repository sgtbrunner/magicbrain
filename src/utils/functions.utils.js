export const getUserFromLocalStorage = () => localStorage.getItem('user');
