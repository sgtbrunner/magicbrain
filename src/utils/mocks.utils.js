export const USER = {
  id: 3,
  name: 'test',
  email: 'test@test.com',
  entries: 7,
};

export const OUTPUTS = {
  outputs: [
    {
      data: {
        regions: [
          {
            region_info: {
              bounding_box: {
                top_row: 0.5,
                left_col: 0.5,
                bottom_row: 0.5,
                right_col: 0.5,
              },
            },
          },
        ],
      },
    },
  ],
};

export const CREDENTIAL_ERROR = {
  error: 'You have entered an invalid username and/or password',
};

export const SERVER_ERROR = {
  error: 'Something went wrong. Please try again later.',
};
