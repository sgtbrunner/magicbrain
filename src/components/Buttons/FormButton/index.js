import FormButton from './form-button.component';

export default FormButton;
