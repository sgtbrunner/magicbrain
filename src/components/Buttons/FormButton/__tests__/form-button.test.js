import { screen } from '@testing-library/react';

import { mountComponent } from '../../../../utils/test.utils';
import FormButton from '..';

const REGISTER = 'Register';
const BUTTON = 'button';

const validClass = 'b bg-lightest-blue grow';

const mountFormButtonComponent = (props) => mountComponent({ component: FormButton, props });

describe('FormButton component', () => {
  it('should match snapshot', () => {
    const { container } = mountFormButtonComponent({ children: REGISTER });

    expect(container).toMatchSnapshot();
  });

  it(`should render 'Register' as button text if children prop is 'Register'`, () => {
    mountFormButtonComponent({ children: REGISTER });

    const button = screen.getByRole(BUTTON);
    expect(button).toBeInTheDocument();
    expect(button).toHaveTextContent(REGISTER);
  });

  it(`should NOT be disabled if disabled prop is false`, () => {
    mountFormButtonComponent({ children: REGISTER, disabled: false });

    const button = screen.getByRole(BUTTON);
    expect(button).toBeEnabled();
  });

  it(`should be disabled if disabled prop is true`, () => {
    mountFormButtonComponent({ children: REGISTER, disabled: true });

    const button = screen.getByRole(BUTTON);
    expect(button).toBeDisabled();
  });

  it(`should NOT have valid form classes if isValidForm is false `, () => {
    mountFormButtonComponent({ children: REGISTER, isValidForm: false });

    const button = screen.getByRole(BUTTON);
    expect(button).not.toHaveClass(validClass);
  });

  it(`should have valid form classes if isValidForm is true `, () => {
    mountFormButtonComponent({ children: REGISTER, isValidForm: true });

    const button = screen.getByRole(BUTTON);
    expect(button).toHaveClass(validClass);
  });
});
