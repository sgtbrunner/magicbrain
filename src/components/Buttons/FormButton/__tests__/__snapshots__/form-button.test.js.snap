// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormButton component should match snapshot 1`] = `
<div>
  <button
    class="ph3 pv2 mt3 ba b--black f6 dib "
    type="submit"
  >
    Register
  </button>
</div>
`;
